javascript/ql/src/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -643,6 +643,60 @@ module TaintTracking {
 
   }
 
+  /**
+   * A less-than or greater-than expression
+   */
+  private class ExclusiveRelationalComparison extends RelationalComparison {
+    ExclusiveRelationalComparison() {
+      this instanceof LTExpr or
+      this instanceof GTExpr
+    }
+  }
+
+  /** 
+   * A check of the form `if(whitelist.indexOf(x) >= 0)`, which sanitizes `x` in its "then" branch. 
+   *
+   * Similar relational checks are also supported.
+   */
+  class RelationalIndexOfSanitizer extends AdditionalSanitizerGuardNode, DataFlow::ValueNode {
+    MethodCallExpr indexOf;
+    override RelationalComparison astNode;
+    boolean polarity;
+
+    RelationalIndexOfSanitizer() {
+      exists (Expr lesser, Expr greater |
+        astNode.getLesserOperand() = lesser and
+        astNode.getGreaterOperand() = greater and
+        indexOf.getMethodName() = "indexOf" |
+        polarity = true and
+        greater = indexOf and
+        (
+          lesser.getIntValue() = 0 
```

Comment:
How about making this `>= 0`? Perhaps it's speculative generality, but on the other hand it's a single-character change.